### PR TITLE
fix: rpc send transfer recipient choose fee

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build a739014 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/8708501028), [Test report](https://leather-wallet.github.io/playwright-reports/fix/send-transfer-recipient), [Storybook](https://fix-send-transfer-recipient--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=fix/send-transfer-recipient)

This pr fixes bug with rpc send transfer fees selection. Here I just duplicated components in order not to affect other flows. In next pr I will refactor it

### CODE REVIEW

**Comment by kyranjamie** [comment]
Thanks @alter-eggo, think we should include in this PR:
- Fixing the styling on the confirmation page, it's not centred properly
- Adding validation if network & network of addresses provided don't match
- Adding validation if decimal amount provided, and we expect sat denominated amounts only

File: src/shared/rpc/methods/send-transfer.ts [hunk 1]
```diff
@@ -19,11 +25,27 @@ export const rpcSendTransferParamsSchemaLegacy = yup.object().shape({
...
+  recipients: yup.array().of(
+    yup.object().shape({
+      // check network is valid for address
+      address: btcAddressValidator().test(
+        'address-network-validation',
+        FormErrorMessages.IncorrectNetworkAddress,
+        (value, context) => {
+          const contextOptions = context.options as any;
+          const network = contextOptions.from[1].value.network as BitcoinNetworkModes;
+          return btcAddressNetworkValidator(network).isValidSync(value);
```
Comment (kyranjamie) [inline src/shared/rpc/methods/send-transfer.ts]: magic number, what's 1?

File: src/app/common/transactions/bitcoin/utils.ts [hunk 2]
```diff
@@ -144,3 +145,108 @@ export function getBitcoinTxValue(address: string, transaction?: BitcoinTx) {
...
+      } else {
+        acc[outputType] = 1;
+      }
+
+      return acc;
+    },
+    {} as Record<AddressType, number>
+  );
+
+  const outputsData = (Object.keys(outputTypesLengthMap) as AddressType[]).map(v => {
```
Comment (kyranjamie) [inline src/app/common/transactions/bitcoin/utils.ts]: what is v? please name fully

File: src/app/common/transactions/bitcoin/utils.ts [hunk 3]
```diff
@@ -144,3 +145,108 @@ export function getBitcoinTxValue(address: string, transaction?: BitcoinTx) {
...
+  });
+  const outputAddressesTypesWithFallback = addressesInfo.map(addressInfo =>
+    addressInfo ? addressInfo.type : AddressType.p2wpkh
+  );
+
+  const outputTypesLengthMap = outputAddressesTypesWithFallback.reduce(
+    (acc: Record<AddressType, number>, outputType) => {
+      // we add 1 output for change address if not sending max
+      if (!acc['p2wpkh'] && !isSendMax) {
+        acc['p2wpkh'] = 1;
```
Comment (kyranjamie) [inline src/app/common/transactions/bitcoin/utils.ts:221]: what is 1?

File: src/shared/rpc/methods/send-transfer.ts (hunk 1 above)
Comment (alter-eggo) [reply src/shared/rpc/methods/send-transfer.ts]: seems, there is no other ways to [get network here ](https://stackoverflow.com/questions/73724613/yup-validation-at-lower-level-based-on-a-higher-value-unsure-how-to-access)

File: src/app/common/transactions/bitcoin/utils.ts (hunk 3 above)
Comment (alter-eggo) [reply src/app/common/transactions/bitcoin/utils.ts:221]: default output length value

File: src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts [hunk 4]
```diff
@@ -96,3 +103,103 @@ export function determineUtxosForSpend({
...
+  const sizeInfo = getSizeInfoMultipleRecipients({
+    inputLength: filteredUtxos.length,
+    isSendMax: true,
+    recipients,
+  });
+
+  // Fee has already been deducted from the amount with send all
+  const outputs = recipients.map(({ address, amount }) => ({ value: BigInt(amount), address }));
+
+  const fee = Math.ceil(sizeInfo.txVBytes * feeRate);
```
Comment (kyranjamie) [inline src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts:140]: use BigNumber for math

File: src/app/common/transactions/bitcoin/utils.ts (hunk 2 above)
Comment (alter-eggo) [reply src/app/common/transactions/bitcoin/utils.ts]: renamed to outputAddressType

File: src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts (hunk 4 above)
Comment (alter-eggo) [reply src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts:140]: most of this fn is copy-pasted, I can refactor math there later?

**Review by kyranjamie — APPROVED** [review]
Thanks @alter-eggo. Let's follow up with tests and refactor as soon as this is shipped!

**Pull request merged**